lib/policyEvaluator/RequestContext.ts

```diff
@@ -15,9 +15,19 @@ import {
     actionMapScuba,
 } from './utils/actionMaps';
 
-const _actionNeedQuotaCheck = {
+export const actionNeedQuotaCheck = {
     objectPut: true,
+    objectPutVersion: true,
     objectPutPart: true,
+    objectRestore: true,
+};
+
+// This variable describe APIs that change the bytes
+// stored, requiring quota updates
+export const actionWithDataDeletion = {
+    objectDelete: true,
+    objectDeleteVersion: true,
```

Comment:
It depends where you take the `apiMethod`, it seems you take it from `request.apiMethod`, in that case, I think there is no `objectDeleteVersion` (https://github.com/scality/Arsenal/blob/development/8.1/lib/s3routes/routes/routeDELETE.ts#L52) it is mixed in the `objectDelete` handler.

The `Version` suffix is added in the RequestContext for Vault https://github.com/scality/cloudserver/blob/development/8.7/lib/api/apiUtils/authorization/prepareRequestContexts.js#L61

Reply:
When evaluating the quotas, we do it for each of the returned actions from the authorization results. Today it's only the action provided as the input, but we might eventually have more actions (see below). These actions can have `Version` suffix.

We use the `request.apiMethod` only at the end, for the monitoring, where it will indeed not be `objectDeleteVersion`. So we need to support them.

Now, for the multi object delete, indeed I think I am not properly handling it in the internal additional calls, when we authorize one by one all the objects we want to delete. I will work on that.

Reply:
Just an update: I added the inflights for the multi delete in cloudserver.